elasticsearch/src/main/scala/com/snowplowanalytics/stream/loader/clients/ElasticsearchBulkSender.scala

```diff
@@ -92,7 +94,19 @@ class ElasticsearchBulkSender(
       .builder(formedHost)
       .setHttpClientConfigCallback(httpClientConfigCallback)
       .setDefaultHeaders(headers)
-    ElasticClient.fromRestClient(restClientBuilder.build())
+    ElasticClient(JavaClient.fromRestClient(restClientBuilder.build()))
+  }
+
+  /**
+   * This BulkHandler is added to change endpoints of bulk api request to add
+   * document type to them. This change is made in order to continue to support ES 6.x
+   */
+  implicit object CustomBulkHandler extends Handler[BulkRequest, BulkResponse] with Logging {
+    override def build(t: BulkRequest): ElasticRequest = {
+      val req = BulkHandlers.BulkHandler.build(t)
+      val ep  = s"${documentType.map(t => s"/$documentIndex/$t").getOrElse("")}/_bulk"
```

Comment:
I suggest we make it very explicit that for 7.x server you are using the default request, and not making any custom changes.  Maybe something like this:

```scala
  implicit object CustomBulkHandler extends Handler[BulkRequest, BulkResponse] with Logging {
    override def build(t: BulkRequest): ElasticRequest = {
      val req = BulkHandlers.BulkHandler.build(t)
      documentType match {
          case None => req
          case Some(t) => req.copy(endpoint = s"/$documentIndex/$t{req.endpoint}")
      }
    }
  }
```